packages/flutter_tools/lib/src/commands/build_ios.dart

```diff
@@ -255,7 +255,8 @@ abstract class _BuildIOSSubCommand extends BuildSubCommand {
 
     if (!result.success) {
       await diagnoseXcodeBuildFailure(result, globals.flutterUsage, globals.logger);
-      throwToolExit('Encountered error while ${xcodeBuildActionToString(xcodeBuildAction)}ing for $logTarget.');
+      final String presentParticiple = xcodeBuildAction == XcodeBuildAction.build ? 'building' : 'archiving';
```

Comment:
I could alternatively make a variant of `xcodeBuildActionToString` with a case to catch future enum values.